### ISSUE TITLE
[[Not an issue, an attempt to get in contact with you]]

### ISSUE THREAD

**Issue by Kenocrystal** [Yoshimaster96/mpgc-sound-tools#1]
Heyo!

I'm trying to get in contact with you about the BMS to MIDI converter that you have created here:
https://kuribo64.net/board/thread.php?id=2363&page=1

The reason for this is because I've been trying to convert the Mario Galaxy 1/2 BMS files to midi for my own personal projects and research. But I have been extremely and frustratingly unsuccessful. I've tried compiling the code myself and I have tried downloading the builds on the forum, but to no avail. I've tried downloading mingw so I could compile it myself - but I keep having installation issues that I cannot fix and no solutions online help me. I've been so burnt out trying to get this to work but I just cannot.

I cannot find any other functioning BMS to MIDI converters out there and any folder that I find that have rips of the Galaxy soundtracks are either incomplete or taken down. I legally own a copy of the game of myself so I don't have to worry about Nintendo Laywers showing up at my doorstep. But I just want to convert these damn files!

Could you please help me out here? I need a build of this program that you have created that will allow me to convert the Mario Galaxy BMS files to MIDI files. I'm out of options and I'm too burnt out and tired trying to attempt to do it myself over and over. Trying not to sound dramatic here, but you are my only hope.

**Comment by Yoshimaster96** [comment]
I'll be honest, I haven't been working on that stuff for quite a while. I'd recommend checking out camthesaxman's converter [here](https://github.com/camthesaxman/bms2mid), as it's much more up-to-date. I am trying to make it even more accurate, but it's WIP and not quite finished yet. Hopefully the tool I linked will suffice for the time being.

Also I'd recommend using MinGW-w64, the original MinGW is pretty outdated.

**Comment by Kenocrystal** [comment]
"Also I'd recommend using MinGW-w64, the original MinGW is pretty outdated."

Yeah no thats what I meant
Thank you though

**Comment by Kenocrystal** [comment]
ahhahahah

Mingw-w64 is needed to compile this, I am on the brink of giving up on everything
I am sorry, I get stressed very easily

Do you have mingw installed? If so could you compile this for me and send it to me via google drive or something?
If you are confused by what I mean - I am talking about the converter that you recommended to me

**Comment by Yoshimaster96** [comment]
Here's how to install it (assuming you're on Windows):

1. Download the archive file and unpack it wherever (you can find it [here](https://github.com/niXman/mingw-builds-binaries/releases)).
2. Add mingw64\bin to the PATH environment variable (how to do this depends on your Windows version, lemme know if you need help with this step).

And that's it! You can now run gcc and compile the program.

**Comment by Kenocrystal** [comment]
I've compiled it and ran it (At least I think I have, there is an exe file that seems to work whenever I put it into the Command Prompt) but I am not sure how to use it correctly. I've tried all sorts of methods and ways to try to convert my bms files but it just doesn't do anything. I've been at this at a solid hour or two and I am getting extremely frustrated.

**Comment by Yoshimaster96** [comment]
If you run the program with no arguments it will tell you how to use it:

`bms2mid bmsFile midFile instrumentList`
bmsFile is the input BMS file
midiFile is the output MIDI file
instrumentList is an optional argument, you can ignore it.

**Comment by Kenocrystal** [comment]
yes, but how do I phrase the arguments? Do I just type out `bms2mid [file name]` then run it?
I'll do some more testing to try to figure it out myself, I'll let you know if I can get it to work or not

**Comment by Kenocrystal** [comment]
Nevermind! I figured out the phrasing of the arguments! But now I've ran into another issue, whenever I attempt to convert the files the output is `Assertion failed: voice < 8, file bms2mid.c, line 229` and it gives me a corrupted midi file

If you want to see the full error here it is
`C:\Users\USERNAME>C:\Users\USERNAME\Downloads\bms2mid-master\bms2mid-master\bms2mid.exe` `"C:\Users\USERNAME\Desktop\Projects\Mario\Mario Galaxy Project\szs\smg_astrodome_midi_cnv.szs"`
`"C:\Users\USERNAME\Desktop\Projects\Mario\Mario Galaxy Project\szs\smg_astrodome_midi_cnv.szs.midi"`
`Assertion failed: voice < 8, file bms2mid.c, line 229`

Don't let the file extension fool you, these are BMS files

**Comment by Yoshimaster96** [comment]
Hm, not sure, will have to look into that specific file and see what's going on. Could be an issue with the program not handling certain events properly, in which case I don't know what else to do other than modifying the program (which as I said is still very much WIP and won't be done for quite a while).

**Comment by Kenocrystal** [comment]
Well I am not experienced enough to code anything or modify any programs myself. All I have is the files and the original archive of the files I need converted. I first thought that the files that the archive gave me weren't bms files, but the archive was located where the game (Super Mario Galaxy) stores the files. And many arc. file decompressors/extractors extract the archive incorrectly or cannot do it at all. I have to use a szs extractor to extract the files from the archive. 

So if you wanna take a gander at it I can send you the individual BMS files and archive if you wanna mess with it yourself or anything like that.

**Comment by Yoshimaster96** [comment]
That's fine, I have the game files and can look at them when I get the chance.

**Comment by Kenocrystal** [comment]
How have things been?

**Comment by Yoshimaster96** [comment]
Doing alright. What do you need?

**Comment by Kenocrystal** [comment]
Just checking up on you, really. Have you had a chance to take a look at the files to see if anything is wrong?

**Comment by Yoshimaster96** [comment]
No I haven't, sorry.

**Comment by Kenocrystal** [comment]
Thats alright, take your time, I'm currently focused on other projects at the moment so its fine.
